python/ray/air/checkpoint.py

```diff
@@ -361,7 +385,15 @@ def _to_directory(self, path: str) -> None:
                 # This used to be a true fs checkpoint, so restore
                 _unpack(data_dict[_FS_CHECKPOINT_KEY], path)
             else:
-                # This is a dict checkpoint. Dump data into checkpoint.pkl
+                # This is a dict checkpoint.
+                # First, restore any additional files
+                additional_files = data_dict.pop(
+                    _DICT_CHECKPOINT_ADDITIONAL_FILE_KEY, {}
+                )
+                for file, content in additional_files.items():
+                    _unpack(stream=content, path=os.path.join(path, file))
```

Comment:
Do we need to create the directories or does tar take care of that?

Reply:
tar takes care of that